components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/DAOFactory.java

```diff
@@ -365,6 +366,11 @@ private static AnalyticsDAO getAnalyticsDaoImplForVendor(Connection connection)
         return analyticsDAO;
     }
 
+    public static SystemApplicationDao getSystemApplicationDao() throws APIMgtDAOException {
+        SystemApplicationDao systemApplicationDao = new SystemApplicationDaoImpl();
+        return systemApplicationDao;
```

Comment:
preferred way would be to create and return in a single line: return new SystemApplicationDaoImpl();